src/pybind/mgr/volumes/fs/operations/volume.py

```diff
@@ -227,7 +247,56 @@ def list_volumes(mgr):
         result.append({'name': f['mdsmap']['fs_name']})
     return result
 
-
+def info(mgr, path, fs_handle, volname):
+    """
+    Get all information of a CephFS volume.
+    """
+    vol_info_dict = {} # type: Dict[str, Any]
+    df = mgr.get("df")
+    pool_stats = dict([(p['id'], p['stats']) for p in df['pools']])
+    osdmap = mgr.get("osd_map")
+    pools = dict([(p['pool'], p) for p in osdmap['pools']])
+    metadata_pool_id, data_pool_ids = get_pool_ids(mgr, volname)
+    vol_info_dict["pools"] = {"metadata" : [], "data" : []}
+    for pool_id in [metadata_pool_id] + data_pool_ids:
+        if pool_id == metadata_pool_id:
+            pool_type = "metadata"
+        else:
+            pool_type = "data"
+        vol_info_dict["pools"][pool_type].append({
+                        'name' : pools[pool_id]['pool_name'],
+                        'used': pool_stats[pool_id]['bytes_used'],
+                        'avail': pool_stats[pool_id]['max_avail']})
+                             
+    mon_addr_lst = []
+    mon_map_mons = mgr.get('mon_map')['mons']
+    for mon in mon_map_mons:
+        ip_port = mon['addr'].split("/")[0]
+        mon_addr_lst.append(ip_port)
+    vol_info_dict["mon_addrs"] = mon_addr_lst
+    try:
+        st = fs_handle.statx(path.encode('utf-8'), cephfs.CEPH_STATX_SIZE,
+                                cephfs.AT_SYMLINK_NOFOLLOW)
+                                
+        usedbytes = st['size']
+        num_groups = len(listdir(fs_handle, path, filter_entries=[Trash.GROUP_NAME.encode('utf-8')]))
```

Comment:
@neesingh-rh I think the CSI team is asking for this as it takes long time doing a list by themselves on cluster with large number of subvolumegroups. I think the same is applicable doing it from 'mgr/volumes' as well. 
Could you please test this out. If it takes too long on thousands of subvolumegroups, we should probably consider taking this out. 

@vshankar What do you think ?

Reply:
@Madhu-1 For what purpose this gets used? Does the number of subvolume[group]s count _really_need to be accurate?

Reply:
@vshankar This will be used by Rook, Rook checks for any subvolumegroup that exists in a filesystem before deleting the filesystem, and also it will check for the subvolumes in a subvolumegroup before deleting it. Yes, the count should be accurate or else we might end up deleting the resources which are getting utilized. some details here https://github.com/ceph/ceph/pull/45949#discussion_r881728391

Reply:
Does the interface needs to return the count of subvolume[groups]? IOW, could the check (interface) be based on a boolean value being returned? We could read one directory entry at a time and then stop when we see a subvolume[group], inferring that the volume/subvolumegroup is not empty. @kotreshhr wdyt? Worst case, this would result in 2-3 calls to the MDS...

Reply:
Yes, make sense to return the bool value to indicate whether its subvolumegroups are present or not.

Reply:
Okay!!

Reply:
Sounds good to me!